ui/selectors/selectors.js

```diff
@@ -2977,8 +2978,22 @@ export function getMetaMetricsDataDeletionStatus(state) {
   return state.metamask.metaMetricsDataDeletionStatus;
 }
 
+/**
+ * Gets the remote feature flags by combining flags from both the manifest and state.
+ * Manifest flags take precedence and will override any duplicate flags from state.
+ * This allows for both static (manifest) and dynamic (state) feature flag configuration.
+ *
+ * @param {object} state - The MetaMask state object
+ * @returns {object} Combined feature flags object with manifest flags taking precedence over state flags
+ */
 export function getRemoteFeatureFlags(state) {
-  return state.metamask.remoteFeatureFlags;
+  const manifestFlags = getManifestFlags().remoteFeatureFlags;
+  const stateFlags = state.metamask.remoteFeatureFlags;
+
+  return {
+    ...stateFlags,
+    ...manifestFlags,
+  };
```

Comment:
This is going to behave unexpectedly if given the following data:

```
  const stateFlags = {
    flag1: { a: 1, b: 2 },
    flag3: false,
  };

  const manifestFlags = {
    flag1: { b: 3 },
    flag2: false,
  };
```

Let's use lodash merge, and let's also add a unit test to test this.

Reply:
thanks for the suggestion ! all added in 248ec01ff147253884c18464451992729007ac1f